util/util_test.go

```diff
@@ -146,3 +147,67 @@ func TestSliceRemoveIfPresent_struct(t *testing.T) {
 		t.Errorf("Slice should be modified")
 	}
 }
+
+func TestParsePath(t *testing.T) {
+	d := schema.TestResourceDataRaw(t, map[string]*schema.Schema{
+		"name": {Type: schema.TypeString},
+	}, map[string]interface{}{
+		"name": "foo",
+	})
+	result := ParsePath("my/transform/hello", "/transform/role/{name}", d)
+	if result != "/my/transform/hello/role/foo" {
+		t.Fatalf("received unexpected result: %s", result)
+	}
+}
+
+func TestLastField(t *testing.T) {
+	testCases := []struct {
+		input    string
+		expected string
+	}{
+		{
+			input:    "/transform/alphabet",
+			expected: "alphabet",
+		},
+		{
+			input:    "/transform/alphabet/{name}",
+			expected: "{name}",
+		},
+		{
+			input:    "/transform/decode/{role_name}",
+			expected: "{role_name}",
+		},
+		{
+			input:    "/transit/datakey/{plaintext}/{name}",
+			expected: "{name}",
+		},
+		{
+			input:    "/transit/export/{type}/{name}/{version}",
+			expected: "{version}",
+		},
+		{
+			input:    "/unlikely",
+			expected: "unlikely",
+		},
+	}
+	for _, testCase := range testCases {
+		actual := LastField(testCase.input)
+		if actual != testCase.expected {
+			t.Fatalf("input: %q; expected: %q; actual: %q", testCase.input, testCase.expected, actual)
+		}
+	}
+}
+
+func TestPathParameters(t *testing.T) {
```

Comment:
This seems like it could use more tests - there are a bunch of paths through `PathParameters` that aren't being taken

Reply:
Sure, that led me to end up adding support for other types of endpoints as well while I was at it. https://github.com/terraform-providers/terraform-provider-vault/pull/746/commits/07bae43f4a67cdca4ab6f7c17f097879e901e674